### PR TITLE
Add examples for VecDeque

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #29348.

r? @steveklabnik

### CODE REVIEW

**Comment by frewsxcv** [comment]
Looks great, nice work! 👍

File: src/libcollections/vec_deque.rs [hunk 1]
```diff
@@ -696,6 +712,25 @@ impl<T> VecDeque<T> {
...
+    /// ```
+    /// use std::collections::VecDeque;
+    ///
+    /// let mut vector: VecDeque<u32> = VecDeque::new();
+    ///
+    /// vector.push_back(0);
+    /// vector.push_back(1);
+    /// vector.push_back(2);
+    ///
+    /// assert_eq!(vector.as_slices(), (&[0u32, 1, 2] as &[u32], &[] as &[u32]));
```
Comment (sinkuu) [inline src/libcollections/vec_deque.rs]: I think this is easier to read and more conventional:
```rust
assert_eq!(vector.as_slices(), (&[0, 1, 2][..], &[][..]));
```

Comment (frewsxcv) [inline src/libcollections/vec_deque.rs]: Another option:

```rust
assert_eq!(vector.as_slices(), (&[0u32, 1, 2] as &[_], &[] as &[_]));
```

Comment (GuillaumeGomez) [inline src/libcollections/vec_deque.rs]: I prefer to be as explicit as possible. I don't see the point to use `&[_]` instead of `&[u32]`.

**Comment by alexcrichton** [comment]
@bors: retry

* #34595

**Pull request merged**